probes/runsStatusChecksBeforeMerging/def.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: runsStatusChecksBeforeMerging
+short: Check that the project runs required status checks
+motivation: >
+  Required status checks can check for common errors and resolve issues in PRs.
+implementation: >
+  The probe checks the rules for each branch in the projects repository.
+outcome:
+  - The probe returns one OutcomePositive for each branch that runs required status checks, and one OutcomeNegative for branches that don't.
+remediation:
+  effort: High
+  text:
+    - Enable required status checks by following [these guidelines](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks).
```

Comment:
if including github doc links, remove the `/en/` portion so they re-direct to user's default locale.
e.g. 

https://docs.github.com/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks

Reply:
https://github.com/ossf/scorecard/pull/3691/commits/c26b8ce39984154c1f07e99397d8d1af86cbee57